Menus/MainMenu/MainMenu.cs

```diff
@@ -273,14 +277,9 @@ void OnButtonHovered(Transform rayOrigin)
 			this.Pulse(this.RequestNodeFromRayOrigin(rayOrigin), m_ButtonHoverPulse);
 		}
 
-		void OnOpening()
-		{
-			this.Pulse(node, m_ShowPulse);
-		}
-
-		void OnClosing()
+		public void SendVisibilityPulse()
```

Comment:
Ah, right. I forgot a crucial thing here. When we're showing the menu, the hideflags are set to 0, so we don't know whether it was because of the activator or because of an auto-hide. That's why SendVisibilityPulse is called by the activator method. The only way I can think to solve this is to keep track of a lastHideFlags state for each menu, which seems awkward.

Thoughts?